### PR TITLE
feat: landing page code highlighting

### PR DESCRIPTION
**Pull request by brendan-defi**
**What changed? Why?**
* Added syntax highlighting to component code snippets

Before (Light):
image

After (Light):
image

Before (Dark):
image

After (Dark):
image

**Notes to reviewers**

**How has it been tested?**
* locally

### CODE REVIEW

**Review by fakepixels — APPROVED** [review]

File: site/docs/components/App.tsx [hunk 1]
```diff
@@ -23,31 +23,10 @@ const wagmiConfig = createConfig({
   },
 });
 
-export default function App({ children }: { children: ReactNode }) {
-  const [theme, setTheme] = useState('light');
```
Comment (cpcramer) [inline site/docs/components/App.tsx:27]: Curious why we are removing theming this level?

Comment (cpcramer) [reply site/docs/components/App.tsx:27]: This breaks components updating based on dark/light mode, see [/swap/swap](https://onchainkit-czk9bf4hw-coinbase-vercel.vercel.app/swap/swap) as an example

This component should be using the light mode theme:

Screenshot 2024-10-24 at 5 27 52 PM

Comment (brendan-defi) [reply site/docs/components/App.tsx:27]: I see, sorry about that. 

The reason I pulled `theme` up to ComponentPreview` level is that I need theme at that level in order to render the code block with the right style. So I thought that I would keep it at that level and just pass as props to `<App>` which lives in `ActiveComponent`s. 

I didn't realize that `<App>` was used elsewhere and had fixed the structure for the files I could find. I see now that it's also used in markdown files. 

Will get this fixed.

File: site/docs/components/landing/CheckoutDemo.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ function CheckoutDemo() {
   }, []);
 
   return (
-    <App>
+    <App theme={theme}>
```
Comment (alessey) [inline site/docs/components/landing/CheckoutDemo.tsx:32]: can revert these now

File: site/docs/components/landing/ComponentPreview.tsx [hunk 3]
```diff
@@ -167,8 +172,16 @@ function PreviewContainer({
   setActiveSubTab,
   copiedIndex,
   copyToClipboard,
+  theme,
 }: PreviewContainerProps) {
   const ActiveComponent = components[activeTab].component;
+  const [highlightedCode, setHighlightedCode] = useState<React.ReactNode>(null);
```
Comment (alessey) [inline site/docs/components/landing/ComponentPreview.tsx:178]: let import ReactNode to be consistent with the other uses

File: site/docs/components/landing/FundDemo.tsx [hunk 4]
```diff
@@ -25,9 +25,9 @@ export const fundDemoCode = `
   <FundButton fundingUrl={onrampBuyUrl} />
 `;
 
-function FundDemo() {
+function FundDemo({ theme }: { theme: string }) {
   return (
-    <App>
+    <App theme={theme}>
```
[the remark this answers is not in the document]
Comment (brendan-defi) [reply site/docs/components/landing/FundDemo.tsx]: done

File: site/docs/components/landing/TransactionDemo.tsx [hunk 5]
```diff
@@ -12,31 +12,31 @@ import App from '../App.tsx';
...
 
   const { address } = useAccount();
   const handleOnStatus = useCallback((status: LifecycleStatus) => {
     console.log('LifecycleStatus', status);
   }, []);
-    
+
   <TransactionDefault
     contracts={contracts}
   />
```
Comment (alessey) [inline site/docs/components/landing/TransactionDemo.tsx:42]: remove whitespace on save eh?

**Review by alessey — APPROVED** [review]
lgtm

File: site/docs/components/landing/TransactionDemo.tsx (hunk 5 above)
Comment (brendan-defi) [reply site/docs/components/landing/TransactionDemo.tsx:42]: ha yeah i guess my default linting rules do that

**Pull request merged**